lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -307,6 +313,9 @@ pub trait SegmentOptimizer {
         ids: Vec<SegmentId>,
         stopped: &AtomicBool,
     ) -> CollectionResult<bool> {
+        let mut timer = TelemetryOperationTimer::new(&self.get_telemetry_counter());
+        timer.set_success(false);
```

Comment:
what is the exact semantic of `success` for a `TelemetryOperationTimer`?

Reply:
The idea is simple. We count fail and success runs. The timer adds value to statistics in `drop` function. If this function return `Ok` at the end of optimization, we set timer as successful. Otherwise, if any `?` return Err, the timer will add failed run to statistics

Reply:
Neat, thanks for the explanation :+1: